### PR TITLE
tools: change build-radosgw.sh to use ceph's repo script

### PR DESCRIPTION
**Pull request by 0xavi0**
This commit goes in conjunction with https://github.com/aquarist-labs/ceph/pull/68

In order to make ceph repo independant from s3gw-tools and be able to run the unittests and other tests the build script has been moved to the ceph repository.

The new script also offer the possibility to set a user defined build script (via the new env variable BUILD_SCRIPT) in case we have special needs for particular tests or to extend the ci on the ceph repository.

If, for example, we need to add a new flag when building the gateway, or a new test it will be done in the ceph repository and the change will be tested in its own ci in the same pull request.

This way the build container remains the same being just a holder of the needed environment, but the how we build the gateways is ruled by the ceph repository.

This also moves the tests to the ceph repository, so we can also run them in any ceph's PR without depending on s3gw-tools.

It also creates 2 new containers for running the `radosgw` and its unit tests in the `ceph` repo CI.

Those containers are mere base for running those processes and the binaries and libraries are mounted. That way we don't need to fully create a container with the binaries and we can reuse them.

Fixes: https://github.com/aquarist-labs/s3gw/issues/127

Signed-off-by: Xavi Garcia <xavi.garcia@suse.com>

# Describe your changes

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] CHANGELOG.md has been updated should there be relevant changes in this PR.

### CODE REVIEW

File: build/build-radosgw.sh [hunk 1]
```diff
@@ -1,126 +1,6 @@
...
-}
-
-build_radosgw
-strip_radosgw
-
-if [ "${WITH_TESTS}" == "ON" ]; then
-  build_radosgw_test
-fi
-
+BUILD_SCRIPT=${BUILD_SCRIPT:-"/srv/ceph/qa/rgw/store/sfs/build-radosgw.sh"}
```
Comment (giubacc) [inline build/build-radosgw.sh:4]: Why an absolute path as default here?
I think we should privilege the user and not the CI on another repository.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added `unittest_rgw_sfs_metadata_compatibility` and `unittest_rgw_sfs_gc` unit tests
   for testing rgw/sfs.
 
+- RGW tests moved to ceph repository.
```
Comment (giubacc) [inline CHANGELOG.md:15]: please, remove blank lines to preserve the existing convention

**Review by giubacc — DISMISSED** [review]

File: build/build-radosgw.sh (hunk 1 above)
Comment (0xavi0) [reply build/build-radosgw.sh]: Done.. using the `CEPH_DIR` env variable now

File: CHANGELOG.md (hunk 2 above)
Comment (0xavi0) [reply CHANGELOG.md:14]: done

**Comment by 0xavi0** [comment]
I tried to test the build process changing to a pull request, but there's a constrain in gh that prevents accessing the secrets.
**_Secrets are not passed to workflows that are triggered by a pull request from a fork._**

I tested the workflow file in my own repository and it worked. here's the build:
https://github.com/0xavi0/s3gw-core/actions/runs/3572191784/jobs/6004801137

**Review by jecluis — DISMISSED** [review]
lgtm

**Review by jecluis — APPROVED** [review]

**Pull request merged**